core/src/main/resources/lib/form/hetero-list/hetero-list.js

```diff
@@ -7,49 +7,49 @@ Behaviour.specify(
   "hetero-list",
   -100,
   function (e) {
-    e = $(e);
     if (isInsideRemovable(e)) {
       return;
     }
 
     // components for the add button
     var menu = document.createElement("SELECT");
-    var btns = findElementsBySelector(e, "INPUT.hetero-list-add"),
-      btn = btns[btns.length - 1]; // In case nested content also uses hetero-list
+    // In case nested content also uses hetero-list
+    var btn = Array.from(e.querySelectorAll("INPUT.hetero-list-add")).pop();
     if (!btn) {
       return;
     }
     YAHOO.util.Dom.insertAfter(menu, btn);
 
-    var prototypes = $(e.lastElementChild);
-    while (!prototypes.hasClassName("prototypes")) {
-      prototypes = prototypes.previous();
+    var prototypes = e.lastElementChild;
+    while (!prototypes.classList.contains("prototypes")) {
+      prototypes = prototypes.previousElementSibling;
     }
-    var insertionPoint = prototypes.previous(); // this is where the new item is inserted.
+    var insertionPoint = prototypes.previousElementSibling; // this is where the new item is inserted.
 
     // extract templates
     var templates = [];
-    var i = 0;
-    $(prototypes)
-      .childElements()
-      .each(function (n) {
-        var name = n.getAttribute("name");
-        var tooltip = n.getAttribute("tooltip");
-        var descriptorId = n.getAttribute("descriptorId");
-        // YUI Menu interprets this <option> text node as HTML, so let's escape it again!
-        var title = n.getAttribute("title");
-        if (title) {
-          title = title.escapeHTML();
-        }
-        menu.options[i] = new Option(title, "" + i);
-        templates.push({
-          html: n.innerHTML,
-          name: name,
-          tooltip: tooltip,
-          descriptorId: descriptorId,
-        });
-        i++;
+    var children = prototypes.children;
+    for (var i = 0; i < children.length; i++) {
+      var n = children[i];
```

Comment:
you could simplify this code and have it more similar to before with:

```suggestion
    Array.from(prototypes.children)
      .forEach(n => {
```

/ with a more explanative variable name:
```suggestion
    Array.from(prototypes.children)
      .forEach(childElement => {
```

Reply:
Seems fine the way it is.